userspace/libsinsp/threadinfo.cpp

```diff
@@ -148,10 +148,6 @@ void sinsp_threadinfo::init()
 	m_exe_writable = false;
 	m_exe_upper_layer = false;
 	m_exe_from_memfd = false;
-
-	memset(&m_user, 0, sizeof(scap_userinfo));
```

Comment:
Don't we need to memset to zero the `sinsp_user` and `sinsp_group` structs now?

Reply:
No, I don't think so. In the way how it's currently implemented, a default constructor will be created for `sinsp_userinfo`/`sinsp_groupinfo` and called here, which in turn will initialize string (as empty strings) and leave built-in typed `uid`/`gid` uninitialized. In contrast, doing `memset` will zero everything, destroying strings internal structure.

The only question here is whether or not to initialize `uid`/`gid` explicitly to zero. I've decided here not to do so, since zero isn't a special value, and could be mixed up with the "root" user or group. But if you folks prefer it to be zero, I can add a constructor to `sinsp_userinfo`/`sinsp_groupinfo`.

Reply:
Oh you are right, that's no more a struct, it's a class now. 

> The only question here is whether or not to initialize uid/gid explicitly to zero

I am pretty sure it should be -1 (UINT32_MAX), like we do here:
https://github.com/falcosecurity/libs/pull/1765/files#diff-f1f5adc2cc836ddaf885f86ded61c940d5c957781ea3ef5cef7f4c6f407054f7R2132

Reply:
Done:
https://github.com/falcosecurity/libs/pull/1765/commits/e4738000d437ac0356a0c11d97965f48e4197b34#diff-42bb444cadccd57c45d52c21568207e3ab49d2d9cd8eb8df92faf4898aed9b3aR71